### PR TITLE
deck: completed function to create and shuffle a UNO card deck.

### PR DESCRIPTION
**Pull request by sethdivyansh**
## Description
1. Added black to CardColor in types.d.ts.
2. Split special and wild cards into specialValues and wildCardValues arrays.
3. For each color in the colors array:
- Add two cards for each number (0-9), except one card for '0'.
- Add two cards for each special value.
- For 'black', add four cards for each wild card value.

Fixes: #1 

## Checklist
- [x] I have tested these changes locally.
- [ ] I have reviewed the code and ensured it follows the project's coding guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have assigned reviewers to this pull request.

### CODE REVIEW

**Comment by kuv2707** [comment]
Can you edit the commit message to occupy less width per line?

**Comment by sethdivyansh** [comment]
> Can you edit the commit message to occupy less width per line?

I have shortened the commit message.

**Comment by kuv2707** [comment]
Great work!
It would be further great if you could add some tests to verify if the array generated has all the cards in the right amount.
Feel free to take the test cases from [this PR](https://github.com/shivansh-bhatnagar18/multiplayer-uno/pull/54/files) if you don't feel like writing it yourself (do credit the author for it though).

File: backend/uno-game-engine/deck.ts [hunk 1]
```diff
@@ -10,12 +10,10 @@ const values = [
...
     '9',
-    'skip',
-    'reverse',
-    'draw2',
 ];
-const specialCards = ['wild', 'draw4'];
-const deck = [];
+const specialValues: Array<SpecialCardNames> = ['skip', 'reverse', 'draw2'];
+const wildCardValue: Array<SpecialCardNames> = ['colchange', 'draw4'];
+const deck: Array<UNOCard> = [];
```
Comment (kuv2707) [inline backend/uno-game-engine/deck.ts:16]: Seems like this variable is useless.
You can go ahead and remove it.

File: backend/uno-game-engine/deck.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const colors: Array<CardColor> = ['red', 'yellow', 'green', 'blue'];
-const values = [
+const numValues: Array<CardNumbers> = [
```
Comment (kuv2707) [inline backend/uno-game-engine/deck.ts:2]: Make CardNumbers singular.

File: backend/uno-game-engine/deck.ts [hunk 3]
```diff
@@ -10,12 +10,10 @@ const values = [
     '7',
     '8',
     '9',
-    'skip',
-    'reverse',
-    'draw2',
 ];
-const specialCards = ['wild', 'draw4'];
-const deck = [];
+const specialValues: Array<SpecialCardNames> = ['skip', 'reverse', 'draw2'];
```
Comment (kuv2707) [inline backend/uno-game-engine/deck.ts:14]: Here too, types should be singular, unless they are an array

File: backend/uno-game-engine/deck.ts [hunk 4]
```diff
@@ -42,11 +40,27 @@ const sameCardCount = []; // to keep track of same cards in assigning unique id
...
+        specialValues.forEach((value) => {
+            deck.push(makeCard('special', color, value));
+            deck.push(makeCard('special', color, value));
+        });
+        if (color === 'black') {
+            wildCardValue.forEach((value) => {
+                for (let i = 0; i < 4; i++) {
+                    deck.push(makeCard('wild', color, value));
+                }
+            });
```
Comment (kuv2707) [inline backend/uno-game-engine/deck.ts:61]: The generation logic seems fine.
You can add tests to check if it generates the entire suite correctly.

File: backend/uno-game-engine/deck.ts [hunk 5]
```diff
@@ -10,12 +10,10 @@ const values = [
...
     '8',
     '9',
-    'skip',
-    'reverse',
-    'draw2',
 ];
-const specialCards = ['wild', 'draw4'];
-const deck = [];
+const specialValues: Array<SpecialCardNames> = ['skip', 'reverse', 'draw2'];
+const wildCardValue: Array<SpecialCardNames> = ['colchange', 'draw4'];
```
Comment (kuv2707) [inline backend/uno-game-engine/deck.ts:15]: The variable name should be plural.

**Review by kuv2707 — CHANGES_REQUESTED** [review]
Posted some minor comments.
Overall, this seems promising and can be merged once the comments are addressed and tests are added.

PS. the commit message is perfect. Kudos for that!

File: backend/uno-game-engine/deck.ts (hunk 1 above)
Comment (kuv2707) [reply backend/uno-game-engine/deck.ts:16]: This would also make the eslint-backend check pass.

**Comment by sethdivyansh** [comment]
@kuv2707 Please check I have made the desired changes and added a test to validate the created function.

File: backend/tests/deck.test.ts [hunk 6]
```diff
@@ -1,10 +1,68 @@
-import { shuffle, makeCard } from '../uno-game-engine/deck';
+import {
+    shuffle,
+    makeCard,
+    getShuffledCardDeck,
+} from '../uno-game-engine/deck';
+
+describe('getShuffledCardDeck', () => {
```
Comment (kuv2707) [inline backend/tests/deck.test.ts:7]: Appreciate the thorough testing!

**Review by kuv2707 — APPROVED** [review]
Looks great!

**Pull request merged**